### ISSUE TITLE
Deceptive instructions for installing Astropy in CASA

### ISSUE THREAD

**Issue by embray** [astropy/astropy#7818]
These docs are a bit outdated and have a few problems: http://docs.astropy.org/en/stable/install.html#installing-astropy-into-casa

1. The suggestion of using `easy_install` should be dropped entirely.
2. Don't suggest using `pip.main` from a Python prompt.  The `pip` module was never intended to be used as a library and the fact that people do so has been a headache for the pip developers.  `pip.main` has since been deprecated and removed (actually moved to `pip._internal`) to discourage its use.
3. I have played with Astropy in CASA before but not in a long time, so I don't know why it's easier to ask users to do things this way.  But if it must be (as opposed to just providing the right `python -m pip ...` command), something like this would be better:

```
import subprocess, sys
subprocess.check_call([sys.executable, '-m', 'pip', 'install', ...])
```

**Comment by astrofrog** [comment]
@keflavich - did you say this doesn't actually work anymore? Time to remove these instructions?

**Comment by keflavich** [comment]
Correct, these instructions fail spectacularly in some cases.  

Related issues:
https://github.com/astropy/astroquery/issues/1219 
https://github.com/radio-astro-tools/casa-python/issues/19
https://stackoverflow.com/questions/52289107/installation-of-astropy-in-casa

**Comment by astrofrog** [comment]
@keflavich - do you agree we should just remove it? If the CASA developers aren't interested in helping support astropy, that's kind of their issue.

**Comment by bsipocz** [comment]
Agreed that this actually needs to be documented in CASA rather than the upstream in astropy.

**Comment by astrofrog** [comment]
Well I mean to be clear they won't document it because they don't support it. So removing it here means it won't be documented anywhere, but things just change too much from version to version for these instructions to be useful.

**Comment by bsipocz** [comment]
but if they don't willing to support it, then what can we do anyway 🤷‍♀️

**Comment by astrofrog** [comment]
When the PR is opened for this, it should be backported to LTS and 3.0.x to make sure the instructions don't pop up anywhere.

**Comment by keflavich** [comment]
I'm opposed to "just removing" these docs.  I really need there to be a clear path to installing into CASA.  The CASA devs never supported *anything* we did, it's not like anything has changed there; the change is that `pip` has more explicitly deprecated the approach we were taking.

I'd prefer to add a loud warning that the prescription listed in the docs does not generally work, and point to the associated issues.

**Comment by astrofrog** [comment]
@keflavich - would you be willing to open a PR for this? 😄 I wonder if we should basically also include a sentence to say 'if you want to see astropy included or more easily installable in CASA, open an issue here' and point to the CASA issue tracker?

**Comment by embray** [comment]
> I'd prefer to add a loud warning that the prescription listed in the docs does not generally work, and point to the associated issues.

Why that, over just changing the prescription to something different that will work just as well if not better than the old one?

**Comment by keflavich** [comment]
because we don't know how yet.  If we had a better solution, we'd put it up, but we haven't found one that works.

**Comment by keflavich** [comment]
because for reasons I don't understand, pip installing astropy that way breaks ipython because of `shutil_get_terminal_size`.

There are several unrelated issues here.  Yes, `subprocess.check_call...` is the right replacement for `pip.main...`, but pip installing astropy in general still doesn't work with CASA without a lot of extra work.

**Comment by embray** [comment]
In that case, I would say that is a problem for CASA and maybe we should remove these instructions entirely if it's not supported.  Or fork CASA :)

**Comment by keflavich** [comment]
We now have at least one example (user test case yesterday) where the `subprocess.check_call` version worked, so we should just update the instructions to do that.

CASA users are part of the astropy user base, though.  We should not consider entirely removing instructions targeted at them.  Yes, this is a CASA problem, but the CASA team has demonstrated that they will generally not help solve these problems.

**Comment by keflavich** [comment]
Moving the documentation off of astropy pages actively pushes away the CASA-using community.  I don't support that.  I want to bring CASA and astropy together, not keep them separated.

**Comment by bsipocz** [comment]
Keeping the pages outdated as a function of CASA releases then in return makes the astropy doc viewed as unreliable and out of date.

**Comment by keflavich** [comment]
This particular deprecation happened in `pip`.  The docs are out of date.  This is no different from astropy having to keep up with other upstream changes, e.g., as in numpy, python itself, etc.  

I can see an argument for not trying to keep up with changes in CASA, though I disagree with it, but so far CASA hasn't made any significant changes in its infrastructure for a long time.  The docs fell out of date mostly because of things unrelated to both astropy and CASA.

**Comment by bsipocz** [comment]
OK, that was a misunderstanding from my side then. 

I still don't like that it's *us* who ends up with providing this info, and looking at the original PR that added the content (https://github.com/astropy/astropy/pull/4840), others felt the same way. But given it went through the first time, I suppose it can stay.

**Comment by keflavich** [comment]
That's fair, these instructions don't feel great as part of the astropy library package, but including them is definitely part of the astropy project's mission.  Including this bit of documentation is an important bit of outreach that at least tries to signal that we're willing to support radio astronomers.

**Comment by embray** [comment]
> This particular deprecation happened in pip.

Yes-and-no.  As far back as I remember the pip developers have said "don't use pip.main".  But it was common enough that they finally had to just break it to get people to stop.

**Comment by embray** [comment]
Why is CASA so resistant to working to integrate better with Astropy?  This seems like a high-level community discussion that needs to happen because this seems like a needless split...

**Comment by keflavich** [comment]
This has been an ongoing (for at least 5 years) discussion.  The answer is layered: bureaucracy, momentum, and CASA is mostly a C++ package with a python frontend.

**Comment by embray** [comment]
I would say "ok so what" r.e. being mostly C++ with a Python frontend.  The biggest challenge seems to have more to do with how it's distributed, as a monolithic, almost IRAF-like system.

**Comment by bsipocz** [comment]
@keflavich - I'm sorry for assigning this to you, but I would like to make sure the instructions are cleaned up by 3.1, and I suppose you're our in-house CASA expert 😉

**Comment by pllim** [comment]
There was a stop-gap warning inserted in #8043 for 3.1, so removing milestone.

**Comment by pllim** [comment]
Fixed in #8185
